spinalcordtoolbox/deepseg/inference.py

```diff
@@ -316,3 +322,45 @@ def segment_nnunet(path_img, tmpdir, predictor, device: torch.device):
         fnames_out.append(fname_out)
 
     return fnames_out, targets
+
+
+def segment_totalspineseg(path_img, tmpdir, predictor, device):
+    # for totalspineseg, the 'predictor' is just the model path
+    path_model = predictor
+
+    # Copy the file to the temporary directory using shutil.copyfile
+    path_img_tmp = os.path.join(tmpdir, os.path.basename(path_img))
+    shutil.copyfile(path_img, path_img_tmp)
+    logger.info(f'Copied {path_img} to {path_img_tmp}')
+
+    # Create directory for nnUNet prediction
+    tmpdir_nnunet = os.path.join(tmpdir, 'nnUNet_prediction')
+    os.mkdir(tmpdir_nnunet)
+
+    tss_inference(
+        # totalspineseg expects pathlib.Path objects
+        input_path=Path(path_img_tmp),
+        output_path=Path(tmpdir_nnunet),
+        data_path=Path(path_model),
+        # totalspineseg expects the device type, not torch.device
+        device=device.type,
+        # The remaining args mimic the default arguments of totalspineseg's argparse
+        output_iso=False,
+        loc_path=None,
+        suffix="",
+        loc_suffix="",
+        step1_only=False,
+        max_workers=os.cpu_count(),
+        max_workers_nnunet=int(max(
+            min(os.cpu_count(),
+                psutil.virtual_memory().total / 2**30 // 8),
+            1
+        )),
+        quiet=False,
+    )
+    fnames_out, targets = [], []
+    for output_dirname in ["step1_canal", "step1_cord", "step1_levels", "step1_output", "step2_output"]:
+        fnames_out.append(os.path.join(tmpdir_nnunet, output_dirname, os.path.basename(path_img)))
+        targets.append(f"_{output_dirname}")
```

Comment:
Regarding the `torch.device` request, we need to update totalspineseg to use nnunet api instead of cli. Is it ok if we keep it as it is for now ?

Regarding outputting a list of the final paths, I am not really sure it is the role of the inference function. However, I printed the paths at the end with a quick description to make sure that people understand the different outputs.